### PR TITLE
Mono font

### PR DESCRIPTION
**Pull request by NickeZ**
Added Courier New

### CODE REVIEW

File: src/workflow/confirm.c [hunk 1]
```diff
@@ -105,11 +105,16 @@ enum workflow_async_ready workflow_confirm_async(
     }
 }
 
-bool workflow_confirm(const char* title, const char* body, bool longtouch, bool accept_only)
+bool workflow_confirm(
+    const char* title,
+    const char* body,
+    bool longtouch,
+    bool monospaced,
```
Comment (benma) [inline src/workflow/confirm.c]: What about providing directly the desired font, like here? https://github.com/digitalbitbox/bitbox02-firmware/blob/4dfc14acc8d38f61e6c5e66c495b0325eddbe9df/src/ui/components/label.c#L286

Can be passed down all the way to the confirm label component, which default to the old one if NULL.

File: src/workflow/confirm.c [hunk 2]
```diff
@@ -105,11 +105,16 @@ enum workflow_async_ready workflow_confirm_async(
     }
 }
 
-bool workflow_confirm(const char* title, const char* body, bool longtouch, bool accept_only)
+bool workflow_confirm(
```
Comment (benma) [inline src/workflow/confirm.c]: We get more and more combinations of the confirm functions and params. Later we could think about making only one function that takes a config struct instead with good defaults, then calling it is more readable.

https://github.com/digitalbitbox/bitbox02-firmware/pull/253 will also need a new combination where the confirmation buttons are at the top during transaction confirmation :sweat_smile:

Comment (benma) [reply src/workflow/confirm.c]: If you do this, then using the password font here would be possible too:
https://github.com/digitalbitbox/bitbox02-firmware/pull/266#issuecomment-567472999

File: src/workflow/confirm.c (hunk 1 above)
Comment (NickeZ) [reply src/workflow/confirm.c]: ack

File: src/workflow/confirm.c (hunk 2 above)
Comment (NickeZ) [reply src/workflow/confirm.c]: sounds like a good refactoring issue. Can you create it?

**Comment by NickeZ** [comment]
I've changed to propagating the `UG_FONT*`. So let's merge this before the password font.

File: src/workflow/confirm.c (hunk 2 above)
Comment (benma) [reply src/workflow/confirm.c]: Done

File: src/workflow/unlock.c [hunk 3]
```diff
@@ -84,19 +84,25 @@ static bool _get_mnemonic_passphrase(char* passphrase_out)
...
                     "Your passphrase starts or ends with a space, or contains consecutive spaces.",
+                    NULL,
                     true)) {
                 return false;
             }
         }
         bool cancel_forced = false;
-        if (workflow_confirm_scrollable_longtouch("Confirm", passphrase_out, &cancel_forced)) {
+        if (workflow_confirm_scrollable_longtouch(
+                "Confirm", passphrase_out, NULL, &cancel_forced)) {
```
Comment (benma) [inline src/workflow/unlock.c]: This should use the password font to make the spaces visible.

Comment (benma) [reply src/workflow/unlock.c]: Nevermind, I'm in the wrong PR.

Comment (NickeZ) [reply src/workflow/unlock.c]: Not in this PR. That comes with the password font PR.

**Comment by thisconnect** [comment]
@NickeZ isn't the Courier New License an issue?

https://docs.microsoft.com/en-us/typography/font-list/courier-new

**Comment by NickeZ** [comment]
Yes obviously we need licenses to modify and redistribute the fonts we use.

**Comment by NickeZ** [comment]
For now we will go with a public domain font instead of purchasing license for Courier New.

**Pull request closed without merging**